astropy/utils/decorators.py

```diff
@@ -1101,3 +1100,87 @@ def set_docstring(obj):
         obj.__doc__ = doc.format(*args, **kwargs)
         return obj
     return set_docstring
+
+
+def prepend_docstring(doc, sections=None):
+    """
+    Decorator to prepend the given docstring to the decorated function's
+    after stripping out the list of sections provided.
+
+    Parameters
+    ----------
+    doc : str
+        Docstring to pretend.
+
+    sections : list or None
+        Name of sections to remove from ``doc`` and replace with the
+        decorated doctring one. Default is None to remove no sections.
+    """
+    def decor(func):
+        func.__doc__ = ("\n".join(_remove_sections(doc, sections)) +
```

Comment:
I wonder of `os.linesep` is more OS-agnostic than `"\n"`, although someone told me once that it doesn't matter. So, please ignore if you want.

Reply:
There are tons of `"\n"` in this file, so I would defer this for a possible future PR that deals with them consistently.